reference/JSDoc-Type-References.md

```diff
@@ -0,0 +1,242 @@
+In Javascript files, the compiler understands Typescript types in
+JSDoc comments like `@type` and `@param`. However, it has additional
+rules to make type resolution work better for Javascript code that
+wasn't written with machine checking in mind. These rules apply only
+in JSDoc in Javascript files, and some can be disabled by setting
+`"noImplicitAny": true`, which the compiler takes as a signal that the
+code is being written with Typescript in mind.
+
+This document first describes the way Javascript type resolution is
+different, because those differences are bigger and more surprising.
+Then it covers differences in name resolution.
+
+## Simple Rewrites ##
+
+The simplest Javascript-only rule is to rewrite the primitive object
+types to real primitives.
+
+Type        | Resolved Type
+------------|--------------
+`Number`    | `number`
+`String`    | `string`
+`Boolean`   | `boolean`
+`Object`    | `any`
+
+The rewrite `Object -> any` is disabled when `"noImplicitAny": true`,
+which the compiler takes as a signal that the code is being written
+with Typescript in mind. Old versions of Typescript also rewrote
+`object -> any`, but stopped in Typescript 3.7.
+
+There are also rewrites of other JSDoc types that are equivalent to
+built-in Typescript types:
+
+Type                  | Resolved Type
+----------------------|--------------
+`Null`                | `null`
+`Undefined`           | `undefined`
+`Void`                | `void`
+`function`            | `Function`
+`Object<string,Type>` | `{ [x: string]: Type }`
+`Object<number,Type>` | `{ [x: number]: Type }`
+
+Finally, `Array` and `Promise` both default their type parameters to
+`any` when no type parameters are provided:
```

Comment:
That's not how it works in JS -- JS defaults to any -- although perhaps it should. You are probably thinking of inference's default, but elsewhere in TS you just get an error, and the resulting variable has type `any`:

```ts
var a: Array
```

Reply:
Yes, I was thinking that would happen only if you're using `noImplicitAny`.  (And an `unknown` seems to me a bit better than an error, since it would lead to other errors which will clarify that you need to say what's in it.)  In any case, it's a tangent...